tests/core.test.js

```diff
@@ -0,0 +1,183 @@
+const _ = require("lodash");
+const rewire = require("rewire");
+
+let core;
+
+const mockPluginMethods = {
+  method1: jest.fn().mockImplementation((client) => client),
+  method2: jest.fn(),
+  method3: jest.fn(),
+};
+const mockAutocompleteFunctions = {
+  auto1: jest.fn(),
+  auto2: jest.fn(),
+  auto3: jest.fn(),
+};
+
+const mockAction = {
+  id: "mockAction",
+  params: [],
+};
+const mockSettings = { id: "mockSettings" };
+
+const mockServiceConstructor = jest.fn();
+
+const mockAutocompleteParams = [{
```

Comment:
Please put all those mocked data into the tests which use them. We need to keep them around in order to know what's going on.